src/lifx/light.js

```diff
@@ -529,4 +529,159 @@ Light.prototype.colorZones = function(startIndex, endIndex, hue, saturation, bri
+ */
+Light.prototype.getDeviceChain = function(callback) {
+  validate.callback(callback, 'light getDeviceChain method');
+
+  const packetObj = packet.create('getDeviceChain', {}, this.client.source);
+  packetObj.target = this.id;
+  const sqnNumber = this.client.send(packetObj);
+  this.client.addMessageHandler('stateDeviceChain', function(err, msg) {
+    if (err) {
+      return callback(err, null);
+    }
+    return callback(null, msg);
+  }, sqnNumber);
+};
+
+/**
+ * Sets Tile Position
+ * @example light.setUserPosition(0, 12, 13, 0, () => {})
+ * @param {Number} tileIndex	unsigned 8-bit integer
+ * @param {Number} userX	32-bit float
+ * @param {Number} userY	32-bit float
+ * @param {Number} reserved	16-bit integer
+ * @param {Function} callback called when light did receive message
+ */
+Light.prototype.setUserPosition = function(tileIndex, userX, userY, reserved, callback) {
+  validate.isUint8(tileIndex, 'setUserPosition', 'tileIndex');
+  validate.isXY(userX, userY, 'setUserPosition', 'user');
+  validate.optionalCallback(callback, 'light setUserPosition method');
+
+  const packetObj = packet.create('setUserPositiion', {
+    tileIndex,
+    userX,
+    userY,
+    reserved: reserved || 0
+  }, this.client.source);
+  packetObj.target = this.id;
+  this.client.send(packetObj, callback);
+};
+
+function defaultOptionsTileState64(options) {
+  const ret = {
+    length: options.length,
+    x: options.x,
+    y: options.y,
+    width: options.width,
+    reserved: options.reserved,
+    duration: options.duration
+  };
+  if (typeof ret.length !== 'number') {
+    ret.length = 64;
+  }
+  if (typeof ret.width !== 'number') {
+    ret.length = 8;
+  }
+  if (typeof ret.x !== 'number') {
+    ret.x = 0;
+  }
+  if (typeof ret.y !== 'number') {
+    ret.y = 0;
+  }
+  if (typeof ret.duration !== 'number') {
+    ret.duration = 8;
+  }
+  return ret;
+}
+
+/**
+ * Requests tile GetTileState64 707
+ *
+ * Get the state of 64 pixels in the tile in a rectangle that has
+ * a starting point and width.
+ * The tileIndex is used to control the starting tile in the chain
+ * and length is used to get the state of that many tiles beginning
+ * from the tileIndex. This will result in a separate response from
+ * each tile.
+ * @param {Number} tileIndex	unsigned 8-bit integer
+ * @param {Object} options - options passed to
+ * @param {Number} options.length	unsigned 8-bit integer (default 64)
+ * @param {Number} options.x	unsigned 8-bit integer (default 0)
+ * @param {Number} options.y	unsigned 8-bit integer (default 0)
+ * @param {Number} options.width	unsigned 8-bit integer (default 8)
+ * @param {Number} options.reserved	unsigned 8-bit integer
+ * @param {Function} callback a function to accept the data
+ */
+Light.prototype.getTileState64 = function(tileIndex, options, callback) {
+  validate.isUint8(tileIndex, 'getTileState64', 'tileIndex');
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
```

Comment:
What's the purpose of the above block?

Reply:
the options are optional  that you could pass on parameter options the callback